action/protocol/staking/handlers.go

```diff
@@ -206,34 +207,34 @@ func (p *Protocol) handleWithdrawStake(ctx context.Context, act *action.Withdraw
 		withdrawWaitTime = _withdrawWaitingTime
 	}
 	if blkCtx.BlockTimeStamp.Before(bucket.UnstakeStartTime.Add(withdrawWaitTime)) {
-		return log, &handleError{
+		return log, nil, &handleError{
 			err:           errors.New("stake is not ready to withdraw"),
 			failureStatus: iotextypes.ReceiptStatus_ErrWithdrawBeforeMaturity,
 		}
 	}
 
 	// delete bucket and bucket index
 	if err := delBucketAndIndex(csm, bucket.Owner, bucket.Candidate, act.BucketIndex()); err != nil {
-		return log, errors.Wrapf(err, "failed to delete bucket for candidate %s", bucket.Candidate.String())
+		return log, nil, errors.Wrapf(err, "failed to delete bucket for candidate %s", bucket.Candidate.String())
 	}
 
 	// update withdrawer balance
 	if err := withdrawer.AddBalance(bucket.StakedAmount); err != nil {
-		return log, &handleError{
+		return log, nil, &handleError{
 			err:           errors.Wrapf(err, "failed to update the balance of withdrawer %s", actionCtx.Caller.String()),
 			failureStatus: iotextypes.ReceiptStatus_ErrInvalidBucketAmount,
 		}
 	}
 	// put updated withdrawer's account state to trie
 	if err := accountutil.StoreAccount(csm, actionCtx.Caller, withdrawer); err != nil {
-		return log, errors.Wrapf(err, "failed to store account %s", actionCtx.Caller.String())
+		return log, nil, errors.Wrapf(err, "failed to store account %s", actionCtx.Caller.String())
 	}
-
 	log.AddAddress(actionCtx.Caller)
-	if p.hu.IsPost(config.Greenland, blkCtx.BlockHeight) {
-		log.AddTopics(bucket.StakedAmount.Bytes())
-	}
-	return log, nil
+
+	// generate withdraw amount log
+	amountlog := newReceiptLog(p.addr.String(), HandleWithdrawAmount, true)
+	amountlog.AddTopics(byteutil.Uint64ToBytesBigEndian(bucket.Index), bucket.StakedAmount.Bytes())
```

Comment:
why should it be an receiptLog, but not an action.Log?

Reply:
yeah, can directly use action.Log